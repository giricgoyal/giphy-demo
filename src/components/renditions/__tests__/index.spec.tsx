import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Renditions, { Props } from '../index'

describe('<Renditions />', () => {
    let container, getAllByTestId

    const defaultProps: Props = {
        renditions: {
            original: {
                url: 'https://www.avengers.com/angry-hulk.gif',
            },
            downsized: {
                webp: 'https://www.avengers.com/angry-hulk.webp',
            },
            hd: {
                mp4: 'https://www.avengers.com/angry-hulk.mp4',
            },
        },
    }

    beforeEach(() => {
        ;({ container, getAllByTestId } = render(<Renditions {...defaultProps} />))
    })

    test('should match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('When an item is selected', () => {
        beforeEach(() => {
            fireEvent.click(getAllByTestId('sidebar-item')[2])
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
