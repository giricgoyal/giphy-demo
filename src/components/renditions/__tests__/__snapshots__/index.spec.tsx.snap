// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Renditions /> When an item is selected should match snapshot 1`] = `
<div>
  <div
    class="renditions"
  >
    <div
      class="renditions__sidebar"
    >
      <div
        class="renditions__sidebar__item "
        data-testid="sidebar-item"
      >
        Original
      </div>
      <div
        class="renditions__sidebar__item "
        data-testid="sidebar-item"
      >
        Downsized
      </div>
      <div
        class="renditions__sidebar__item active"
        data-testid="sidebar-item"
      >
        Hd
      </div>
    </div>
    <div
      class="renditions__body"
    >
      <video
        autoplay=""
        loop=""
        src="https://www.avengers.com/angry-hulk.mp4"
      />
    </div>
  </div>
</div>
`;

exports[`<Renditions /> should match snapshot 1`] = `
<div>
  <div
    class="renditions"
  >
    <div
      class="renditions__sidebar"
    >
      <div
        class="renditions__sidebar__item active"
        data-testid="sidebar-item"
      >
        Original
      </div>
      <div
        class="renditions__sidebar__item "
        data-testid="sidebar-item"
      >
        Downsized
      </div>
      <div
        class="renditions__sidebar__item "
        data-testid="sidebar-item"
      >
        Hd
      </div>
    </div>
    <div
      class="renditions__body"
    >
      <img
        src="https://www.avengers.com/angry-hulk.gif"
      />
    </div>
  </div>
</div>
`;
